automata/base/automaton.py

```diff
@@ -8,7 +8,7 @@
 import automata.base.config as global_config
 import automata.base.exceptions as exceptions
 from automata.base.utils import freeze_value
-from typing import Any, Mapping, NoReturn, AbstractSet, Tuple, Dict, Generator
+from typing import Any, Mapping, NoReturn, AbstractSet, Tuple, Dict, Generator, Self
```

Comment:
@caleb531 this is 3.11 only, you need to do `from typing_extensions import Self`.

Reply:
@eliotwrobson Ha yep, was just figuring that out after I noticed the failing build. Glad to know it's in `typing_extensions`. Thanks for the tip.